CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 project(cCOSMO)
-cmake_minimum_required(VERSION 3.0)
-set (CMAKE_CXX_STANDARD 11)
+cmake_minimum_required(VERSION 3.10)
+
+# Set the C++ standard to C++17
+set (CMAKE_CXX_STANDARD 17)
+set (CMAKE_CXX_STANDARD_REQUIRED ON)
+set (CMAKE_CXX_EXTENSIONS OFF)
```

Comment:
Changed minimum required CMake version from 3.0 to 3.10 to leverage new features and improvements in the newer version and set C++ standard to C++17 to ensure compatibility with modern C++ features.